experimental/uefi/runtime/src/remote_attestation.rs

```diff
@@ -22,12 +22,31 @@
 extern crate alloc;
 
 use alloc::vec::Vec;
-use anyhow::Context;
-use oak_remote_attestation_sessions::{SessionId, SessionState, SessionTracker};
+use anyhow::{bail, Context};
+use oak_remote_attestation_sessions::{SessionId, SessionState, SessionTracker, SESSION_ID_LENGTH};
 
 /// Number of sessions that will be kept in memory.
 const SESSIONS_CACHE_SIZE: usize = 10000;
 
+fn deserialize_msg(msg: Vec<u8>) -> anyhow::Result<(SessionId, Vec<u8>)> {
+    let mut session_id: [u8; SESSION_ID_LENGTH] = [0; SESSION_ID_LENGTH];
+    let mut request_body: Vec<u8> = Vec::with_capacity(msg.len() - SESSION_ID_LENGTH);
+
+    if msg.len() < SESSION_ID_LENGTH {
+        bail!("Message too short to contain a session_id.");
+    }
+
+    for (index, byte) in msg.into_iter().enumerate() {
+        if index <= SESSION_ID_LENGTH {
+            session_id[index] = byte;
+        } else {
+            request_body.push(byte);
+        }
+    }
```

Comment:
Updated this as part of improving readability as suggested by @conradgrobler. Not sure if cloning at once is more performant however, even with it effectively just iterates: https://doc.rust-lang.org/std/vec/struct.Vec.html#method.extend_from_slice.

It would be nice to split without reallocating, but it seems that rust doesn't let you do that because of how it handles memory: 
https://stackoverflow.com/a/68180453
https://users.rust-lang.org/t/split-owned-vec-t-without-reallocation/6346